config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/PolicyEntityBuilder.java

```diff
@@ -469,6 +470,32 @@ private void prepareAssertion(Element policyElement, String assertionTag, Consum
         }
     }
 
+    @VisibleForTesting
+    void prepareMQRoutingAssertion(Document policyDocument, Bundle bundle, Element assertionElement) {
+        final Element activeConnectorNameElement = getSingleChildElement(assertionElement, ACTIVE_CONNECTOR_NAME, true);
+        if (activeConnectorNameElement != null) {
+            final String activeConnectorName = activeConnectorNameElement.getAttributes().getNamedItem(STRING_VALUE).getTextContent();
+            final SsgActiveConnector ssgActiveConnector = bundle.getSsgActiveConnectors().get(activeConnectorName);
+            final String id = ssgActiveConnector != null && ssgActiveConnector.getAnnotatedEntity() != null && ssgActiveConnector.getAnnotatedEntity().getId() != null ?
+                    ssgActiveConnector.getAnnotatedEntity().getId() : idGenerator.generate();
+            Element activeConnectorGoidElement = createElementWithAttribute(
+                    policyDocument,
+                    ACTIVE_CONNECTOR_GOID,
+                    GOID_VALUE,
+                    id
+            );
+            assertionElement.insertBefore(activeConnectorGoidElement, activeConnectorNameElement);
```

Comment:
@uppoju , ID and GOID both are required ?

Reply:
yes